### DISCUSSION TITLE
TypeError: callback is not a function

### DISCUSSION THREAD

**Discussion by prati-singh** [webpack/webpack#15014] [Q&A]
# Bug report

**What is the current behavior?**

Getting below error while running webpack.config.js

> [webpack-cli] TypeError: callback is not a function
> at emit (.../node_modules/service-worker-plugin/index.js:54:5)
> at Hook.eval [as callAsync] (eval at create (.../node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), :18:1)
> at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (.../node_modules/webpack/node_modules/tapable/lib/Hook.js:18:14)
> at Compiler.emitAssets (.../node_modules/webpack/lib/Compiler.js:873:19)
> at .../node_modules/webpack/lib/Watching.js:181:21
> at processTicksAndRejections (internal/process/task_queues.js:75:11)
> npm ERR! code ELIFECYCLE
> npm ERR! errno 2
> npm ERR! exp@1.1.0 build-assets-devmode: npm run prepare-webpack && npm run i18n:bundle-client && gulp build-assets:third-party && webpack-cli --watch --config webpack/webpack.config.babel.js
> npm ERR! Exit status 2
> npm ERR!
> npm ERR! Failed at the exp@1.1.0 build-assets-devmode script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR! .../.npm/_logs/2021-12-17T12_46_20_962Z-debug.log

**If the current behavior is a bug, please provide the steps to reproduce.**

1. Updated webpack, webpack-cli, loaders and plugins to their latest version
2. Running webpack is throwing this error
3. Issue happening for [service-worker-plugin](https://www.npmjs.com/package/service-worker-plugin) and at [this](https://github.com/zackargyle/service-workers/blob/master/packages/service-worker-plugin/index.js#L54) point

**What is the expected behavior?**
After installing webpack (5.65.0) latest version, there should not be any error while running webpack.config.js.

Please let me know if any other details are needed.
Thank You!!

**Other relevant information:**
webpack version:  5.65.0
Node.js version:  14.15.4
Operating System: macOS Big Sur 11.6.1
Additional tools:

**Comment by alexander-akait** [comment]
Please create reproducible test repo

**Reply by alexander-akait** [reply]
Looks like they do not support webpack v5, please open an issue in their repo

**Reply by alexander-akait** [reply]
And you should do not use this plugin due https://github.com/zackargyle/service-workers#-this-project-is-no-longer-maintained-, please migrate on another solution

**Reply by prati-singh** [reply]
Thanks @alexander-akait, I will find some alternative plugin for this.
